packages/starknet-snap/src/rpcs/executeTxn.test.ts

```diff
@@ -0,0 +1,190 @@
+      invocations.forEach((invocation) => {
+        invocation.type = transactionType;
+      });
+    }
+    const request: ExecuteTxnParams = {
+      chainId,
+      address,
+      invocations,
+      details,
+    } as ExecuteTxnParams;
+    if (enableAuthorize) {
+      request.enableAuthorize = enableAuthorize;
+    }
+    return request;
+  };
+
+  it('executes transaction correctly when account is deployed', async () => {
+    const account = await mockAccount(constants.StarknetChainId.SN_SEPOLIA);
+    invocationExample = invocationExamples.shift();
+
+    prepareMockAccount(account, state);
+    prepareConfirmDialog();
+    const { createAccountStub, executeTxnRespMock, getEstimatedFeesStub } =
+      prepareMockExecuteTxn(invocationExample.hash, true);
+
+    const request = createRequestParam(
+      state.networks[0].chainId as any,
+      account.address,
+      invocationExample.invocations,
+      invocationExample.details,
+    );
+
+    const result = await executeTxn.execute(request);
+
+    expect(result).toStrictEqual(executeTxnRespMock);
+    expect(executeTxnUtil).toHaveBeenCalledWith(
+      expect.anything(),
+      account.address,
+      account.privateKey,
+      request.invocations.map((invocation) => invocation.payload),
+      undefined,
+      expect.objectContaining({
+        maxFee: '1000000000000000',
+        nonce: undefined,
+      }),
+    );
+    expect(getEstimatedFeesStub).toHaveBeenCalled();
+    expect(createAccountStub).not.toHaveBeenCalled();
+  });
+
+  it('executes transaction correctly when account is not deployed', async () => {
+    const account = await mockAccount(constants.StarknetChainId.SN_SEPOLIA);
+    invocationExample = invocationExamples.shift();
+
+    prepareMockAccount(account, state);
+    prepareConfirmDialog();
+    const { createAccountStub, executeTxnRespMock, getEstimatedFeesStub } =
+      prepareMockExecuteTxn(invocationExample.hash, false);
+
+    const request = createRequestParam(
+      state.networks[0].chainId as any,
+      account.address,
+      invocationExample.invocations,
+      invocationExample.details,
+    );
+
+    const result = await executeTxn.execute(request);
+
+    expect(result).toStrictEqual(executeTxnRespMock);
+    expect(getEstimatedFeesStub).toHaveBeenCalled();
+    expect(createAccountStub).toHaveBeenCalledTimes(1);
+  });
+
+  it('throws error if invocations are empty', async () => {
+    const account = await mockAccount(constants.StarknetChainId.SN_SEPOLIA);
+    prepareMockAccount(account, state);
+    prepareConfirmDialog();
+    prepareMockExecuteTxn(invocationExample.hash, true);
+
+    const request = createRequestParam(
+      state.networks[0].chainId as any,
+      account.address,
+      invocationExample.invocations,
+      invocationExample.details,
+    );
+
+    request.invocations = [];
+
+    await expect(executeTxn.execute(request)).rejects.toThrow(
+      'Invocations cannot be empty',
+    );
+  });
+
+  it.each([
```

Comment:
we dont need that, as starknet is not accept invoke function with declare/deploy/deploy account